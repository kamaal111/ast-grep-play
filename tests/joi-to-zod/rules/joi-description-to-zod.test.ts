import { test, expect } from 'vitest';
import { Lang, parseAsync } from '@ast-grep/napi';

import joiDescriptionToZod from '../../../src/joi-to-zod/rules/joi-description-to-zod';

test('Joi description to Zod describe', async () => {
  const source = `
import Joi from 'joi';

export const employee = Joi.object().keys({
  nickname: Joi.string()
    .required()
    .min(3)
    .max(20)
    .description('Nickname')
    .regex(/^[a-z]+$/, { name: 'alpha', invert: true }),
});
`;
  const ast = await parseAsync(Lang.TypeScript, source);

  const modifications = await joiDescriptionToZod({
    ast,
    report: { changesApplied: 0 },
    lang: Lang.TypeScript,
    filename: null,
    history: [ast],
  });
  const updatedSource = modifications.ast.root().text();

  expect(modifications.report.changesApplied).toBe(1);
  expect(modifications.history.length).toBe(2);
  expect(updatedSource).not.contain('description');
  expect(updatedSource).toMatchSnapshot();
});
