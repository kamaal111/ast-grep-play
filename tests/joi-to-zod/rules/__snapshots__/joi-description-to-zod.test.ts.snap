// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Joi description to Zod describe 1`] = `
"import Joi from 'joi';

export const employee = Joi.object().keys({
  nickname: Joi.string()
    .required()
    .min(3)
    .max(20)
    .describe('Nickname')
    .regex(/^[a-z]+$/, { name: 'alpha', invert: true }),
});
"
`;
